go/host/client_api_eth.go

```diff
@@ -43,14 +43,22 @@ func (api *EthereumAPI) GetBalance(_ context.Context, encryptedParams common.Enc
 	return gethcommon.Bytes2Hex(encryptedBalance), nil
 }
 
-// GetBlockByNumber is a placeholder for an RPC method required by MetaMask/Remix.
-func (api *EthereumAPI) GetBlockByNumber(context.Context, rpc.BlockNumber, bool) (map[string]interface{}, error) {
-	result := map[string]interface{}{
-		// TODO - Return non-dummy values.
-		"baseFeePerGas": (*hexutil.Big)(big.NewInt(0)),
-		"number":        (*hexutil.Big)(big.NewInt(0)),
+// GetBlockByNumber returns the rollup with the given height as a block. No transactions are included.
+func (api *EthereumAPI) GetBlockByNumber(_ context.Context, number rpc.BlockNumber, _ bool) (map[string]interface{}, error) {
+	extRollup := api.host.EnclaveClient.GetRollupByHeight(uint64(number))
+
+	block := map[string]interface{}{
```

Comment:
there's a field missing which we have in the header:
- transactionsroot

Reply:
Think I've added it now. Is it `extRollup.Header.TxHash`?